src/io/result_io.jl

```diff
@@ -400,54 +400,51 @@ function setup_result_store!(domain::Domain, scen_spec::DataFrame)::Tuple
         return (dl...,)
     end
 
-    met_names = [
-        :relative_cover,
-        :relative_shelter_volume,
-        :absolute_shelter_volume,
-        :relative_juveniles,
-        :juvenile_indicator,
-        :coral_evenness
+    outcome_metrics::Vector{metrics.Metric} = [
+        metrics.relative_cover,
+        metrics.relative_shelter_volume,
+        metrics.absolute_shelter_volume,
+        metrics.relative_juveniles,
+        metrics.juvenile_indicator,
+        metrics.coral_evenness,
+        metrics.relative_taxa_cover
     ]
 
-    _unique_loc_ids = unique_loc_ids(domain)
-
-    dim_struct = Dict(
-        :structure => string.((:timesteps, :locations, :scenarios)),
-        :unique_loc_ids => _unique_loc_ids
+    metric_symbols::Vector{Symbol} = metrics.to_symbol.(outcome_metrics)
+    metric_names::Vector{String} = metrics.to_string.(outcome_metrics; is_titlecase=true)
+    metric_units::Vector{String} = [metric.unit for metric in outcome_metrics]
```

Comment:
Don't feel like you need to accept this, but you can use the same broadcasting sugar as the above two lines here too.

```suggestion
    metric_units::Vector{String} = getfield.(outcome_metrics, :unit)
```